src/main/java/com/miraisolutions/xlconnect/ResultWithAttributes.java

```diff
@@ -0,0 +1,42 @@
+package com.miraisolutions.xlconnect;
+
+
+import java.util.*;
+
+/**
+ * Represent attributes to be set on an object in R as part of the result to be returned.
+ * Should be extended for each required type. Not using a generic typed value, because it looks like R can't retrieve it
+ * in a specific subtype (we get an Object instance).
+ */
+class ResultWithAttributes {
```

Comment:
Could we have a generic class here? I mean not in terms of the attributes but in terms of the type that is being decorated, e.g.
```
class AttributeDecorator<T> {
  private T decoratedObject;
  private Map<String, String[]> attributes;

  ...
}
```

Reply:
If for some reason still necessary for interfacing with R, we could still do 
```
class XWithAttributes extends AttributeDecorator<X> { ... }
```

Reply:
In terms of constructor I would have the decorated value first and the attributes after.

Reply:
I kept the same name as it seems fine, but otherwise implemented your proposal. I think I had issues with a generic solution in a previous implementation attempt, but now I define explicitly which functions expect attributes in the result on the R side. No additional aliases seem to be required.